File: ruby/aa/caesar_cipher.rb
def caesar_cipher(str, shift)
  res = ""
  str.each_byte do |byte|
    if byte != " ".ord
      new_byte = (byte - "a".ord + shift) % 26 + "a".ord
      new_char = new_byte.chr
      res << new_char
    else
      res << " "
    end
  end

  res
end

p caesar_cipher("hello kitty", 3)

File: js/aa/caesarCipher.js
function caesarCipher(str, shift){
  let res = "";
  for (let i = 0; i < str.length; i++) {
    if (str[i] !== " ") {
      let newCharCode = ((str[i].charCodeAt() - "a".charCodeAt()) + shift) % 26 + "a".charCodeAt()
      let newChar = String.fromCharCode(newCharCode);
      res = res.concat(newChar);
    } else {
      res = res.concat(" ");
    }
  }
  return res;
}

console.log(caesarCipher("hello kitty", 3));
console.log(caesarCipher("zzx", 3));
